api/base/elasticsearch_dsl_views.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+import abc
+
+import elasticsearch_dsl as edsl
+from rest_framework import generics
+from rest_framework import exceptions as drf_exceptions
+from rest_framework.settings import api_settings as drf_settings
+
+from api.base.filters import FilterMixin
+from api.base.views import JSONAPIBaseView
+
+
+class ElasticsearchListView(FilterMixin, JSONAPIBaseView, generics.ListAPIView, abc.ABC):
+    '''abstract view class using `elasticsearch_dsl.Search` as a queryset-analogue
```

Comment:
Might want to point out when to use this, how this improves on earlier ES6 views.